### ISSUE TITLE
Step is broken when Firebase is installed via SPM

### ISSUE THREAD

**Issue by valerio-bettini** [andrewmarmion/bitrise-step-firebase-dsym-upload#16]
As per subject

**Comment by andrewmarmion** [comment]
This is related to this issue https://github.com/firebase/firebase-ios-sdk/issues/6864. Once that is resolved then I will look at a solution for the step.

**Comment by mikeberlin** [comment]
It looks like the related issue was marked resolved. Any update on fixing this step?

**Comment by andrewmarmion** [comment]
@mikeberlin unfortunately the current solution that has been proposed is more of a work around. I will look into it further, but I am open to PRs.

**Comment by DesmanLead** [comment]
We had to set `-derivedDataPath` for `xcodebuild` to solve this. `SourcePackages` path is determined by the derived data path then.

**Comment by Jack-Warris** [comment]
Hey all, was there ever a clear fix for this? Im having trouble using the route in the linked issue and so can't get this step to pass

**Comment by andrewmarmion** [comment]
Currently this step only supports CocoaPods. To support SPM will require additional work.

**Comment by iamraafay** [comment]
Any update on this?

The ticket referenced above is closed as the suggestions their unblock us from locating the utility well, I can confirm. But this step is still broken and requires a fix to be unblocked on the CI.

**Comment by andrewmarmion** [comment]
@mohdabdurraafay  As I mentioned before this step only supports CocoaPods, but I am open to PRs if it does not break existing functionality.

**Comment by oliverfoggin** [comment]
Hmm... just experiencing this issue also. Has there been an update on this? Thanks

**Comment by andrewmarmion** [comment]
No. This step only supports CocoaPods and does not support SPM.

**Comment by hari-bikki-two-bulls** [comment]
I have figured out a hack to automate this process for the Bitsrise pipeline. I have found a script that will upload the dSYM files when you use SPM.

Copy the `upload-symbols` file and paste it into the below-listed location. 

`$FIREBASE_CRASH_ANALYTICS_LOCATION -gsp $GOOGLE_SERVICE_LOCATION -p ios $BITRISE_DSYM_PATH`

**GOOGLE_SERVICE_LOCATION:** $BITRISE_SOURCE_DIR/PROJECT/Config/GoogleService/GoogleService-Info.plist
**FIREBASE_CRASH_ANALYTICS_LOCATION:** $BITRISE_SOURCE_DIR/PROJECT/Scripts/Firebase/upload-symbols
**BITRISE_DSYM_PATH:** Auto generated by bitrise

References:
https://sarunw.com/posts/upload-dsym-to-firebase-with-fastlane/#copy-upload-symbols-script
https://firebase.google.com/docs/crashlytics/get-deobfuscated-reports?platform=ios&authuser=0&hl=en#upload-dsyms

![image](https://user-images.githubusercontent.com/98289638/206619023-8c9dce83-9fd0-42cd-97ee-ec6e862f7897.png)
![image](https://user-images.githubusercontent.com/98289638/206618713-283358c2-5842-4772-bf72-be13f897a404.png)

**Comment by hari-bikki-two-bulls** [comment]
Update: 

When I pushed the change to Testflight I could not bundle the script file to be deployed.

So Instead of pasting the script file to the project, I added the script to Bitrise GENERIC FILE STORAGE and passed in the URL link and added a file downloader before the script and it was able to fetch the file.

![Screenshot 2022-12-15 at 2 03 57 pm](https://user-images.githubusercontent.com/98289638/207762753-6280074d-8902-4390-a9c6-a25b0f0fc673.png)
![Screenshot 2022-12-15 at 2 04 57 pm](https://user-images.githubusercontent.com/98289638/207762878-52cd7f14-b341-4b74-a3b8-47e5213e8b0b.png)
![Screenshot 2022-12-15 at 2 05 25 pm](https://user-images.githubusercontent.com/98289638/207762931-632b06be-adc6-4e23-9afa-e43ce29b279c.png)

> Error: Invalid bundle structure. The “APP NAME/upload-symbols” binary file is not permitted. Your app cannot contain standalone executables or libraries, other than a valid CFBundleExecutable of supported bundles. For details, visit: https://developer.apple.com/documentation/bundleresources/placing_content_in_a_bundle With error code STATE_ERROR.VALIDATION_ERROR.90171 for id XXXXX Asset validation failed (-19208)
 {
    NSLocalizedDescription = "Invalid bundle structure. The \APPNAME.app/upload-symbols\U201d binary file is not permitted. Your app cannot contain standalone executables or libraries, other than a valid CFBundleExecutable of supported bundles. For details, visit: https://developer.apple.com/docu
mentation/bundleresources/placing_content_in_a_bundle With error code STATE_ERROR.VALIDATION_ERROR.90171 for id XXXXX";
    NSLocalizedFailureReason = "Asset validation failed";
}

**Comment by Daniebro5Bluetread** [comment]
Solution from @hari-bikki-two-bulls worked for me!

**Comment by menjoo** [comment]
Also got it working like this:
Add `upload-symbols` executable somewhere in the repo
Have a Bitrise script step that executes 
```
<path to upload-symbols> -gsp <path to GoogleService-Info.plist> -p ios $BITRISE_DSYM_PATH
```
